packages/extension/src/providers/polkadot/networks/interlay/libs/assetinfo-orml.ts

```diff
@@ -0,0 +1,160 @@
+      let assetLookupValue: string | null = null;
+
+      if (assetKey[AssetIds.FOREIGN_ASSET]) {
+        assetLookupId = "foreignAsset";
+        assetLookupValue = assetKey[AssetIds.FOREIGN_ASSET] as string;
+      } else if (assetKey[AssetIds.NATIVE_ASSET]) {
+        assetLookupId = Object.keys(
+          assetKey[AssetIds.NATIVE_ASSET]
+        )[0] as "token";
+
+        assetLookupValue = (assetKey[AssetIds.NATIVE_ASSET] as NativeAsset)[
+          assetLookupId
+        ] as string;
+      } else if (assetKey[AssetIds.STABLE_ASSET]) {
+        assetLookupId = "stableAssetPoolToken";
+        assetLookupValue = assetKey[AssetIds.STABLE_ASSET] as string;
+      } else if (assetKey[AssetIds.ERC20_ASSET]) {
+        // TODO add Erc20 support, required special RPC call
+      }
+
+      if (assetLookupId && assetLookupValue) {
+        const assetInfo = {
+          name: assetMetadata.name,
+          symbol: assetMetadata.symbol,
+          decimals,
+          minimalBalance,
+          assetLookupId,
+          assetLookupValue,
+        };
+
+        return assetInfo;
+      } else {
+        // Unhandled token types, right now just Erc20
+        return null;
+      }
+    })
+    .filter((asset) => asset !== null);
+
+  const tokenOptions: InterlayOrmlAssetOptions[] = assets
+    .map((asset) => {
+      const ormlOptions: InterlayOrmlAssetOptions = {
+        name: hexToString(asset!.name),
+        symbol: hexToString(asset!.symbol),
+        existentialDeposit: asset!.minimalBalance,
+        assetType: asset!.assetLookupId,
+        lookupValue: asset!.assetLookupValue,
+        icon: network.icon,
+        decimals: asset!.decimals,
+      };
+
+      return ormlOptions;
+    })
+    .map((tokenOptions) => {
+      if (knownTokens) {
+        const knownToken = knownTokens.find(
+          (knownToken) =>
+            knownToken.name === tokenOptions.name &&
+            knownToken.symbol === tokenOptions.symbol
+        );
+
+        if (knownToken) {
+          tokenOptions.coingeckoID = knownToken.coingeckoID;
+          tokenOptions.icon = knownToken.icon;
+        }
+      }
+      return tokenOptions;
+    });
+
+  const nativeAsset = new SubstrateNativeToken({
+    name: network.currencyNameLong,
+    symbol: network.name,
+    decimals: network.decimals,
+    existentialDeposit: network.existentialDeposit,
+    icon: network.icon,
+    coingeckoID: network.coingeckoID,
+  });
+
+  if (address) {
+    await nativeAsset.getLatestUserBalance(apiPromise, address);
+    const queries = tokenOptions.map((asset) => {
+      const token = { [asset.assetType]: asset.lookupValue };
+      const query = [address, token];
+
+      return query;
+    });
+
+    const tokenBalances = await apiPromise.query.tokens.accounts.multi(queries);
+
+    tokenBalances.forEach((tokenData, index) => {
+      const data = tokenData as unknown as OrmlTokensAccountData;
+
+      tokenOptions[index].balance = data.free.toString();
```

Comment:
This would only give transferable balances and exclude the reserved balances. I guess that is OK but just double checking.